### PR TITLE
[RDBMS] `az mysql flexible-server create/update`: Fix the logic of `--public-access` parameter

### PR DESCRIPTION
**Pull request by Ruiion**
**Related command**

We fix the logic of parameter **public-access** for the server create/update command. This parameter is enum. 
- az mysql flexible-server create/update --public-access [Enabled/Disabled]

**Description**

This PR uses latest Python SDK package which in turn uses the latest GA API version 2022-09-30. This PR will allow customers to enable/disabled public access when creating and updating servers.

**Testing Guide**

See examples in _helptext_mysql.py.

enable/disabled public access when creating and updating servers.

**Testing Guide**

See examples in _helptext_mysql.py.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

- [ ] I adhere to the [Error Handling Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/error_handling_guidelines.md).

### CODE REVIEW

**Comment by yonzhan** [comment]
RDBMS

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py [hunk 1]
```diff
@@ -641,7 +650,8 @@ def flexible_server_provision_network_resource(cmd, resource_group_name, server_
         raise RequiredArgumentMissingError("Private DNS zone can only be used with private access setting. Use vnet or/and subnet parameters.")
     else:
         start_ip, end_ip = prepare_public_network(public_access, yes=yes)
-
+        if public_access is not None and str(public_access).lower() == 'Disabled'.lower():
+            network.public_network_access = 'Disabled'
```
Comment (hzhang87) [inline src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py:654]: Please DO NO fully rely on the default value of network.public_network_access from RP. Please explicitly specify network.public_network_access = 'Enabled' when public access is enabled.

File: src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py [hunk 2]
```diff
@@ -641,7 +650,8 @@ def flexible_server_provision_network_resource(cmd, resource_group_name, server_
         raise RequiredArgumentMissingError("Private DNS zone can only be used with private access setting. Use vnet or/and subnet parameters.")
     else:
         start_ip, end_ip = prepare_public_network(public_access, yes=yes)
-
+        if public_access is not None and str(public_access).lower() == 'Disabled'.lower():
```
Comment (hzhang87) [inline src/azure-cli/azure/cli/command_modules/rdbms/flexible_server_custom_mysql.py:653]: I am little bit confused for these two lines of code. Is network.public_network_access case-sensitive or not? Why line 566-567 are made by 
if public_access.lower() == "disabled" or public_access.lower() == "enabled":
            instance.network.public_network_access = public_access
But these two lines are different, while they are in same purpose.

File: src/azure-cli/azure/cli/command_modules/rdbms/validators.py [hunk 3]
```diff
@@ -442,7 +442,8 @@ def ip_address_validator(ns):
 def public_access_validator(ns):
     if ns.public_access:
         val = ns.public_access.lower()
-        if not (val == 'all' or val == 'none' or (len(val.split('-')) == 1 and _validate_ip(val)) or
+        if not (ns.public_access == 'Disabled' or ns.public_access == 'Enabled' or
+                val == 'all' or val == 'none' or (len(val.split('-')) == 1 and _validate_ip(val)) or
                 (len(val.split('-')) == 2 and _validate_ip(val))):
             raise CLIError('incorrect usage: --public-access. '
```
Comment (hzhang87) [inline src/azure-cli/azure/cli/command_modules/rdbms/validators.py:448]: Please also fix raise CLIError by adding 'Enabled' and 'Disabled'

**Review by hzhang87 — APPROVED** [review]
Please fix the comments before merge.

**Comment by evelyn-ys** [comment]
seems duplicate with #26178

**Pull request closed without merging**